ajd/My-List-Util/lib/My/List/Util.pm

```diff
@@ -0,0 +1,165 @@
+package My::List::Util;
+
+use 5.006;
+use strict;
+use warnings;
+
+=head1 NAME
+
+My::List::Util - Some basic list utilities
+
+=head1 VERSION
+
+Version 0.01
+
+=cut
+
+our $VERSION = '0.01';
+
+
+=head1 SYNOPSIS
+
+Provides some list utilities
+
+    use My::List::Util;
+
+    my $foo = My::List::Util->sum( 1..5 );
+    # $foo = 15;
+
+=head1 SUBROUTINES/METHODS
+
+=head2 sum
+
+When supplied with a numeric list, sum() returns the sum total. Any
+non-numeric values are disregarded as 0;
+
+=cut
+
+sub sum {
+
+  my $class = shift @_;
+  my $sum = 0;
+
+  foreach my $element ( @_ ) {
+    # I could get a module to do this check, but I don't want to
+    if ( $element =~ /^-?\d+(?:\.\d+)?$/ ) {
```

Comment:
This won't handle the case where perl stringifies the number using scientific notation - ran into this same thing when adding tests to mine.